### PR TITLE
User should see login or checkout: closes #17

### LINKED ISSUE #17: 10: Guest user has to login to checkout
As a visitor when I have items in my cart 
And when I visit "/cart" 
I should not see an option to "Checkout" 
I should see an option to "Login or Create Account to Checkout" 
After I create an account 
And I visit "/cart 
Then I should see all of the data that was there when I was not logged in

### PR DESCRIPTION
**Pull request by MikelSage**
Adds a button to the cart that prompts visitor to login or create an account that changes to 'Checkout' once they are logged in. Checkout button goes nowhere at the moment.

### CODE REVIEW

File: app/views/carts/index.html.erb [hunk 1]
```diff
@@ -6,24 +6,34 @@
       <div class="col-md-2">
         <img src="<%= item.image %>" class="img-thumbnail">
       </div>
-      <div class="col-md-2">
+      <div class="col-md-1">
         <p><%= item.title %></p>
       </div>
-      <div class="col-md-2">
+      <div class="col-md-5">
```
Comment (jimszalew) [inline app/views/carts/index.html.erb:12]: I like this here, more room for the descriptions

File: app/views/carts/index.html.erb [hunk 2]
```diff
@@ -6,24 +6,34 @@
...
+  <div class="row cart-item">
+    <div class="col-md-2 pull-right">
+      <p>Total Price: $<%= @cart.total_price %></p>
+      <% if current_user %>
+        <%= link_to 'Checkout', '#', class: 'btn btn-primary' %>
+      <% else %>
+        <%= link_to 'Login or Create Account to Checkout', login_path, class: 'btn btn-primary' %>
+      <% end %>
+    </div>
+  </div>
```
Comment (jimszalew) [inline app/views/carts/index.html.erb:38]: Nice work on separating out the views for current_user vs someone who isn't logged in

**Review by jimszalew — APPROVED** [review]
Looks good, nice work!

File: spec/features/cart/visitor_cannot_checkout_spec.rb [hunk 3]
```diff
@@ -10,9 +10,9 @@
     click_button "Add to Cart", match: :first
     page.find("#item-#{item3.id}").click
     click_on "View Cart"
-
-    expect(page).to have_button 'Login or Create Account to Checkout'
-    click_button 'Login or Create Account to Checkout'
```
Comment (dphilla) [inline spec/features/cart/visitor_cannot_checkout_spec.rb]: This looks good, clears up what's needed for checkout.

**Review by dphilla — APPROVED** [review]

File: app/views/carts/index.html.erb (hunk 2 above)
Comment (dphilla) [reply app/views/carts/index.html.erb:38]: I agree with Jim--Looks good!

**Review by dphilla — APPROVED** [review]

**Review by dphilla — APPROVED** [review]
Nice work with the link "login or create account to checkout." solid.

**Pull request merged**